src/pages/o/[orgId]/events/[eventId].tsx

```diff
@@ -52,91 +53,94 @@ type OrgEventPageProps = {
     orgId: string;
 };
 
-export default function OrgEventPage(props : OrgEventPageProps) : JSX.Element {
+export default function OrgEventPage(props : OrgEventPageProps) : JSX.Element | null {
     const { orgId, eventId } = props;
     const eventQuery = useQuery(['event', eventId], getEvent(orgId, eventId));
     const orgQuery = useQuery(['org', orgId], getOrg(orgId));
-    const eventQueryData = eventQuery.data;
 
-    return (
-        //TODO: Break out eventinfo-component and test placeholder text when missing info?
-        <Flex direction="column">
-            <Heading data-test="event-title" level={ 1 }>
-                { eventQueryData?.title ? eventQueryData.title : eventQueryData?.activity.title }
-            </Heading>
-            <Flex>
-                <Text marginEnd="size-50">
-                    <Msg id="pages.orgEvent.details.org"/>:
-                </Text>
-                <Link>
-                    <NextLink href={ `/o/${orgId}` }>
-                        <a>{ orgQuery.data?.title }</a>
-                    </NextLink>
-                </Link>
-            </Flex>
-            <Flex>
-                <Text marginEnd="size-50">
-                    <Msg id="pages.orgEvent.details.campaign"/>:
-                </Text>
-                <Link>
-                    <NextLink href={ `/o/${orgId}/campaigns/${eventQueryData?.campaign.id}` }>
-                        <a>{ eventQueryData?.campaign.title }</a>
-                    </NextLink>
-                </Link>
-            </Flex>
-            <Flex>
-                <Text marginEnd="size-50">
-                    <Msg id="pages.orgEvent.details.start"/>:
-                </Text>
-                { eventQueryData ? (
+    if (!eventQuery.data) {
+        return null;
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    else if (eventQuery.data! && orgQuery.data!) {
```

Comment:
This `else if` should not be necessary. The function returns on line 62 if there is no event so it should be safe to proceed here without doing an `else if`.

Besides, you are asserting (using `!`) that the two data attributes are not `null`. If you know that, why do the `else if`? If you're not sure they're not `null`, you should not be doing the non-null assertion.

If you need to check for `orgQuery.data` as well as `eventQuery.data`, you should do so in the `if` condition on line 61.

At this point, once both org and event queries have been verified as valid, you can safely store the event and org data in new variables, and use type assertion (`as`) to cast them as `ZetkinEvent` and `ZetkinOrganization`, and use those variables going forward.

Reply:
Thanks! Yes, reading this it's clear to me how and why the `else if` is unnecessary.